File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool cmp(vector<int> &v1,vector<int> &v2){
    return v1[0]<v2[0];
}

int solution(vector<vector<int>> targets) {
    int answer = 0;
    sort(targets.begin(),targets.end(),cmp);
    
    int x = targets[0][1];
    int cnt = 1;
    
    for(int i=1;i<targets.size();i++){
        if(x <= targets[i][0]){
            cnt++;
            x= targets[i][1];
        }
        else if(x > targets[i][1])
            x = targets[i][1];
    }
    
    answer = cnt;
    return answer;
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java

import java.io.*;
import java.util.*;
class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        int n = targets.length;
        xy[] v = new xy[n];
        for(int i = 0; i < n; i++){
            v[i] = new xy();
            v[i].s = targets[i][0];
            v[i].e = targets[i][1];
        }
        
        Arrays.sort(v, (a,b)->a.s-b.s);
        int x = v[0].e;
        int cnt = 1;
        
        for(int i = 0; i < n; i++){
            if(x <= v[i].s){
                cnt++;
                x = v[i].e;
            }
            else if(x > v[i].e) x = v[i].e;
        }
        return cnt;
    }
    static class xy{
        int s, e;
    }
}